finesse/gui/measure_script/script_view.py

```diff
@@ -41,6 +42,13 @@ def __init__(self) -> None:
 
         run_btn = QPushButton("Run script")
         run_btn.clicked.connect(self._run_btn_clicked)
+        run_btn.setEnabled(False)
+
+        self._enable_counter = EventCounter(
+            2, lambda: run_btn.setEnabled(True), lambda: run_btn.setEnabled(False)
+        )
+        """A counter to enable/disable the "Run" button."""
+        self._enable_counter.change_on_device_open(STEPPER_MOTOR_TOPIC)
```

Comment:
I would do this step as part of the initialisation of the `EventCounter`, so adding another argument to the constructor with the channel(s) this should be subscribed to and calling internally this method.

Reply:
That makes sense.

Note that you don't *have* to use the device topics with it though. In one place I'm increasing/decreasing the counter depending on the EM27's reported status. It probably does make sense to subscribe to the serial-related messages in the constructor though.